.github/workflows/integration.yml

```
@@ -250,24 +255,50 @@ jobs:
           sudo rm -rf "$AGENT_TOOLSDIRECTORY"
           echo "=== After cleanup:"
           df -h
+
       - uses: actions/checkout@v4
+        with:
+          path: ./agoric-sdk
+
       - name: docker build (sdk)
         # Produces ghcr.io/agoric/agoric-sdk:unreleased used in the following upgrade test.
         # run: cd packages/deployment && ./scripts/test-docker-build.sh | $TEST_COLLECT
         # XXX skip TAP test output and collection for now; it hides the output from the logs
         run: make docker-build-sdk
-        working-directory: packages/deployment
-      - uses: ./.github/actions/restore-node
+        working-directory: agoric-sdk/packages/deployment
+
+      - id: restore-golang
+        uses: ./agoric-sdk/.github/actions/restore-golang
+        with:
+          go-version: '${{ env.GOLANG_VERSION }}'
+          path: ./agoric-sdk
+
+      - id: restore-node
+        uses: ./agoric-sdk/.github/actions/restore-node
         with:
-          node-version: '18.x'
           # Rebuilding the SDK image with resolved endo packages is not currently supported
           # and since we can't build core eval submissions from the SDK using a newer endo,
           # simply ignore any endo branch integration (this means we don't have full coverage)
           ignore-endo-branch: 'true'
-        id: restore-node
+          node-version: '18.x'
+          path: ./agoric-sdk
+          # Forces xsnap to initialize all memory to random data, which increases
+          # the chances the content of snapshots may deviate between validators
+          xsnap-random-init: '1'
+
       - name: setup a3p-integration
         run: yarn install
-        working-directory: a3p-integration
+        working-directory: agoric-sdk/a3p-integration
+
+      - id: build-cosmic-swingset
+        name: Build cosmic-swingset dependencies
+        run: |
+          set -o errexit
+
+          yarn install
+          make --directory packages/cosmic-swingset all
```

I didn't pay close attention to the commands here, where do these come from?
I'd expect the following:

```suggestion
          cd packages/cosmic-swingset
          make install
```

In particular there should be no need to redo `yarn install` as that was done as part of the `restore-node` step. Given caches these are mostly equivalent, just surprised to see something different than `deployment-test`

---

Fixed